aspnetcore/fundamentals/app-state.md

```diff
@@ -263,12 +263,21 @@ public class MyController : Controller
 } 
 ```
 
-### Common errors when working with session
+## Common errors when working with session
 
 * "Unable to resolve service for type 'Microsoft.Extensions.Caching.Distributed.IDistributedCache' while attempting to activate 'Microsoft.AspNetCore.Session.DistributedSessionStore'."
 
   This is usually caused by failing to configure at least one `IDistributedCache` implementation. For more information, see [Working with a Distributed Cache](xref:performance/caching/distributed) and [In memory caching](xref:performance/caching/memory).
 
+* In the event that the session middleware fails to persist a session (for example: if the database is not available), it logs the exception and swallows it. The request will then continue normally, which obviously leads to very unpredictable behavior.
+
+  A typical example:
+
+  Someone stores a shopping basket in session. The user adds an item but the commit fails. The app doesn't know about this so it will happily report a "The item has been added" message, which obviously isn't true.
```

Comment:
- In the 3rd sentence, clarify what "this" means. Should it say something like "The app doesn't know about the commit failure"?
- Remove the words "happily" and "obviously"